Ej16.rb

```diff
@@ -0,0 +1,74 @@
+puts 'Welcome'
+puts 'Cajero Automatico'
+puts 'Ingrese el valor que necesita retirar'
+n=gets.to_i
+bcincuentamil=0
+bveintemil=0
+bdiezmil=0
+bcincomil=0
+bmil=0
+mquinientos=0
+mdoscientos=0
+mcien=0
+mcincuenta=0
+
+cajero=n
+
+while cajero>=50000
```

Comment:
loops are very costly for the processor. 
Try to get rid of the whiles using math expressions. example
`cajero/50_000`
Also, as you can notice, Ruby is a very cool language that lets you put a `_` to separate the thousands so reading big numbers is easier. Please consider using it.